src/App.tsx

```diff
@@ -1,29 +1,89 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import { PaginationInfo } from './components/PaginationInfo';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const items: number[] = getNumbers(1, 42).map(n => n);
+
+const perPageOptions = [3, 5, 10, 20].map((el, i) => {
+  return { id: i, val: el };
+});
+
+interface PageItem {
+  id: number;
+  val: number;
+}
 
 export const App: React.FC = () => {
+  function buildPagesContainer(
+    itms: number[],
+    chunkSize: number,
+  ): PageItem[][] {
+    const contArr: PageItem[][] = [];
+    let globId = 0;
+
+    if (!chunkSize) {
+      return [];
+    }
+
+    for (let i = 0; i < itms.length; i += chunkSize) {
+      const chunk = itms.slice(i, i + chunkSize);
+      const subCont = [];
+
+      for (let i2 = 0; i2 < chunk.length; i2++) {
+        subCont.push({ id: globId, val: chunk[i2] });
+        globId++;
+      }
+
+      contArr.push(subCont);
+    }
+
+    return contArr;
+  }
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(perPageOptions[1].val);
+  const pagesContainer = buildPagesContainer(items, perPage);
+  const currentPageContainer = pagesContainer[currentPage - 1];
+
+  function handlePageSelection(pageNum: number): void {
+    if (pageNum && pageNum <= pagesContainer.length) {
+      setCurrentPage(pageNum);
+    }
+  }
+
+  function handlePagePerSelection(event: React.ChangeEvent<HTMLSelectElement>) {
+    setCurrentPage(1);
+    setPerPage(parseInt(event.target.value));
+  }
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
-      <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
-      </p>
+      <PaginationInfo
+        total={items.length}
+        currentPage={currentPage}
+        firstItem={currentPageContainer[0].val}
+        lastItem={currentPageContainer[currentPageContainer.length - 1].val}
```

Comment:
Be cautious with potential `undefined` values when accessing `currentPageContainer[0].val` and `currentPageContainer[currentPageContainer.length - 1].val`. If `currentPageContainer` is empty, this will result in a runtime error.